crates/nu-parser/src/parser.rs

```diff
@@ -950,6 +950,9 @@ pub fn parse_internal_call(
 
         // Parse a positional arg if there is one
         if let Some(positional) = signature.get_positional(positional_idx) {
+            if positional.name == "def_name" {
```

Comment:
I don't know if we want to make this name magical. What if the user creates a parameter with the name `def_name`?

Reply:
``` python
def provideStr [def_name: record] {
  "def_name_should_print"
}

def needStr [def_name: string] {
  echo $def_name
}

def run [def_name: record] {
  needStr (provideStr $def_name)
}

run {a:b}
```

I tested this code , it throws an error as the real `def_name` shape is String and will detect as mismatch but as the variables are define as `var_id` , `def_name` have None var_id and we can use it to detect which is the real `def_name` , last commit , this example will work

Reply:
Consider to review it please or even any other thing can close it .
Thanks @jntrnr